### PR TITLE
bpo-42317: Improve docs of typing.get_args concerning Union

### PR DESCRIPTION
**Pull request by Dominik1123**
https://bugs.python.org/issue42317

Automerge-Triggered-By: GH:gvanrossum

### CODE REVIEW

File: Doc/library/typing.rst [hunk 1]
```diff
@@ -1706,6 +1706,8 @@ Introspection helpers
    For a typing object of the form ``X[Y, Z, ...]`` these functions return
    ``X`` and ``(Y, Z, ...)``. If ``X`` is a generic alias for a builtin or
    :mod:`collections` class, it gets normalized to the original class.
+   If ``X`` is a :class:`Union`, the order of ``(Y, Z, ...)`` may be different
```
Comment (gvanrossum) [inline Doc/library/typing.rst]:
```suggestion
   If ``X`` is a :class:`Union` contained in another generic type, the order of ``(Y, Z, ...)`` may be different
```

**Comment by Fidget-Spinner** [comment]
I don't really like the current proposed change because it's too broad, it may cause users to worry that _all_ nested types are affected, when in reality it's only Union. 

Instead of:
> If `X` is a generic type, the returned objects `(Y, Z, ...)` might not be identical to the ones used in the form `X[Y, Z, ...]` due to type caching.

Guido's change seems a lot more apt:
> If ``X`` is a :class:`Union` contained in another generic type, the order of ``(Y, Z, ...)`` may be different

A quick search through the typing code tells me that only the `__hash__` and `__eq__` methods of Union are order independent, no other typing type will suffer from this behavior when nested in another Generic.

**Comment by Dominik1123** [comment]
> I don't really like the current proposed change because it's too broad, it may cause users to worry that _all_ nested types are affected, when in reality it's only Union.

It's as broad as the promise of type caching goes, so I don't think it's too broad. But I agree that it's broader than necessary since indeed, at the moment, it requires a `Union` to be involved. Initially I thought it would also affect `Literal`, but apparently it doesn't:

    >>> Literal[1, 2] == Literal[2, 1]
    False

(Though I don't see why the two would be unequal; but that's a different concern)

> Instead of:
> 
> > If `X` is a generic type, the returned objects `(Y, Z, ...)` might not be identical to the ones used in the form `X[Y, Z, ...]` due to type caching.
> 
> Guido's change seems a lot more apt:
> 
> > If `X` is a :class:`Union` contained in another generic type, the order of `(Y, Z, ...)` may be different

So I just revert back to Guido's version?

**Comment by Fidget-Spinner** [comment]
> (Though I don't see why the two would be unequal; but that's a different concern)

Wow good catch, that looks like it may be a bug.

> So I just revert back to Guido's version?

Maybe a combination of both yours and Guido's wording:

```
If `X` is a :class:`Union` contained in another generic type, the order of `(Y, Z, ...)` may be different due to type caching.

Thank you for the PR!
```

**Review by Fidget-Spinner — APPROVED** [review]

**Review by gvanrossum — APPROVED** [review]
Thanks, LG!

**Pull request merged**